### PR TITLE
[REVIEW] Fix importing `apply` from `dask`

### PR DESCRIPTION
**Pull request by galipremsagar**
As part of https://github.com/dask/dask/pull/7801/, `apply` has been removed from `dask.compatibility`, hence made changes in `dask_cudf` to import `apply` from `dask.utils`, this change is back-ward compatible to older versions of dask too.

### CODE REVIEW

**Review by quasiben — APPROVED** [review]
Thanks @galipremsagar !

**Comment by quasiben** [comment]
@gpucibot merge

**Comment by galipremsagar** [comment]
rerun tests

**Comment by galipremsagar** [comment]
rerun tests

**Review by jakirkham — APPROVED** [review]

File: ci/benchmark/build.sh [hunk 1]
```diff
@@ -75,10 +75,10 @@ conda install "rmm=$MINOR_VERSION.*" "cudatoolkit=$CUDA_REL" \
...
 
 # Install the master version of dask, distributed, and streamz
-logger "pip install git+https://github.com/dask/distributed.git@main --upgrade --no-deps"
-pip install "git+https://github.com/dask/distributed.git@main" --upgrade --no-deps
-logger "pip install git+https://github.com/dask/dask.git@main --upgrade --no-deps"
-pip install "git+https://github.com/dask/dask.git@main" --upgrade --no-deps
+logger "pip install git+https://github.com/dask/distributed.git@2021.06.0 --upgrade --no-deps"
+pip install "git+https://github.com/dask/distributed.git@2021.06.0" --upgrade --no-deps
+logger "pip install git+https://github.com/dask/dask.git@2021.06.0 --upgrade --no-deps"
+pip install "git+https://github.com/dask/dask.git@2021.06.0" --upgrade --no-deps
```
Comment (jakirkham) [inline ci/benchmark/build.sh:81]: Why is this needed?

Comment (galipremsagar) [reply ci/benchmark/build.sh:81]: Running benchmarks would fail with latest dask as `streamz` is currently broken: https://github.com/python-streamz/streamz/blob/master/streamz/dask.py#L7

Plan is to revert all the pinning once https://github.com/python-streamz/streamz/pull/423 is merged. - The current pin is to unblock CI

**Review by ajschmidt8 — APPROVED** [review]
